ui-react/packages/atlasmap-core/src/react/Atlasmap.tsx

```diff
@@ -235,9 +239,18 @@ export const Atlasmap: FunctionComponent<IAtlasmapProps> = ({
         onShowMappingPreview={toggleMappingPreview}
         onShowMappedFields={toggleShowMappedFields}
         onShowUnmappedFields={toggleShowUnmappedFields}
+        onConditionalMappingExpressionEnabled={
+          onConditionalMappingExpressionEnabled
+        }
+        onGetMappingExpressionStr={onGetMappingExpressionStr}
+        onToggleExpressionMode={() => {
+          onToggleExpressionMode();
+          expressionTokens = [''];
```

Comment:
Uhm, usually this in the context of a React component is an error. There is no need to "reset" the expressionTokens, if that's the goal. Once the rendering function returns any volatile variable will be lost.